src/core_landice/mode_forward/mpas_li_calving.F

```diff
@@ -1249,6 +1256,794 @@ subroutine eigencalving(domain, err)
+         endwhere
+             
+         !alpha(:) = sigmamin(:) / (sigmamax(:))
+         !epsmax(:) = sigmamax(:) / (2.0d0*visc(:))
+         !szero(:) = rhoi*(rhoo-rhoi)*gravity*thickness(:)/(2.0d0*sigmamax(:)*rhoo)
+          ! Compute the hydrostatic to tensile stress ratio
+          ! These commented lines are doing the same thing as the above where loop. Which is better?
+
+         nstar(:) = 12.0d0*(1.0d0+alpha(:)+alpha(:)**2)/(4.0d0*(1.0d0+alpha(:)+alpha(:)**2)+6.0d0*alpha(:)**2) 
+          ! Compute the effective flow law exponent
+
+         !szero(:) = rhoi*(rhoo-rhoi)*gravity*thickness(:)/(2.0d0*sigmamax(:)*rhoo)
+          ! Compute the hydrostatic to tensile stress ratio
+
+         !do iCell = 1, nCells
+         !    if (li_mask_is_floating_ice(cellMask(iCell))) then
+         !        floatingBasalMassBal(iCell) = 0.2/31536000.0*tanh((lowerSurface(iCell)-bedTopography(iCell))/75.0) * max(-100-lowerSurface(iCell),0.0)
+         !    end if
+         !end do
+         ! Temporary change of BMB for the MISMIP+ Ice1r experiment
+
+         do iCell = 1, nCells
+         if (thickness(iCell) == 0.0_RKIND) then
+             source(iCell) = 0.0_RKIND
+         else
+             source(iCell) = nstar(iCell) * (1.0d0-szero(iCell))*epsmax(iCell)-floatingBasalMassBal(iCell)/rhoi/(thickness(iCell))
+         endif
+         enddo
+
+         seconds = daysSinceStart*24.0*3600.0
+
+         ! below is the force for the manufacured experiments
+         !force(:) = -0.44/2.0 * (uReconstructX(1,:)/1000.0*(1+epsmax(:)*seconds)*exp(-uReconstructX(1,:)*seconds/1000.0) +    &
+         !    source(:) * (1+exp(-uReconstructX(1,:)*seconds/1000.0)))
+         !force(:) = -0.44/2.0 * (uReconstructX(1,:)/1000.0*exp(-uReconstructX(1,:)*seconds/1000.0) +    &
+         !    source(:) * (1+exp(-uReconstructX(1,:)*seconds/1000.0)))
+
+         ! set force to zero if not manufactured
+         force(:) = 0.0
+
+         ddamagedt(:) = source(:) * damage(:) + force(:)
+
+         damage(:) = damage(:) + ddamagedt(:) * deltat
+
+         !damage(:) =  0.5
+         ! temporarily set damage to a constant value
+
+         do iCell = 1, nCells
+         if (thickness(iCell) == 0.0_RKIND) then
+             damage_nye(iCell) = 0.0_RKIND
+         else
+             damage_nye(iCell) = (2.0+alpha(iCell)) * sigmamax(iCell) / (gravity*(rhoo-rhoi)*thickness(iCell))
+         endif
+         enddo
+         !damage_nye(:) = 0.5
+         ! temporarily set damage_nye to a constant value, 0.8
+
+         do iCell = 1, nCells
+            if (damage(iCell) > config_damage_restore_threshold) then
+                damage_max(iCell) = damage(iCell)
+            end if
+         end do
+         ! save the damage_max value for restoring (no heal) later if we want
+
+         do iCell = 1, nCells
+            if (damage(iCell) < 0.0_RKIND) then
+                damage(iCell) = 0.0_RKIND
+            end if
+            if (damage(iCell) < damage_nye(iCell)) then
```

Comment:
Do we always want to do this?

Reply:
I am afraid the answer is yes

Reply:
As per earlier discussion, Jeremy suggests we always want damage to be >= the Nye value. But in that case do we need both if constructs here? It seems like you could omit the first one (if <0) since the latter should always catch either case (if <0 or if <Nye should both be caught by if < Nye).

Reply:
Ok, just wanted to make sure that was part of the theory and not a temporary fix for something.  @tongzhangice , can you add a comment here in the code explaining why this is happening?

Reply:
A comment is added